sdks/java/core/src/main/java/com/google/cloud/dataflow/sdk/runners/worker/IsmFormat.java

```diff
@@ -840,59 +745,29 @@ protected long getEncodedElementByteSize(KeyPrefix value, Coder.Context context)
    *   <li>0x01 (version key as a single byte)</li>
    * </ul>
    */
-  static class Footer {
+  @AutoValue
+  abstract static class Footer {
     static final int LONG_BYTES = 8;
     static final int FIXED_LENGTH = 3 * LONG_BYTES + 1;
     static final byte VERSION = 2;
 
-    private final long indexPosition;
```

Comment:
We could make version an actual field, its not a big deal to store the extra byte for each footer.

Then this would remove even more code from here.